trinity/plugins/builtin/peer_discovery/events.py

```diff
@@ -0,0 +1,36 @@
+from typing import (
+    Tuple,
+    Type,
+)
+
+from lahja import (
+    BaseEvent,
+    BaseRequestResponseEvent,
+)
+
+from p2p.kademlia import (
+    Node,
+)
+
+
+class BaseDiscoveryServiceResponse(BaseEvent):
+
+    def __init__(self, error: Exception) -> None:
+        self.error = error
+
+
+class PeerCandidatesResponse(BaseDiscoveryServiceResponse):
+
+    def __init__(self, candidates: Tuple[str, ...], error: Exception=None) -> None:
+        super().__init__(error)
+        self.candidates = candidates
+
+
+class GetPeerCandidatesRequest(BaseRequestResponseEvent[PeerCandidatesResponse]):
```

Comment:
Mirroring my comment here.  I think these events and the actual `DiscoveryService` can still live in the `p2p` codebase.